server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3231,7 +3231,7 @@ private EngineConfig newEngineConfig(LongSupplier globalCheckpointSupplier) {
             }
         };
         final List<ReferenceManager.RefreshListener> internalRefreshListener;
-        if (indexSettings.isSegrepEnabled()) {
+        if (indexSettings.isSegrepEnabled() && shardRouting.primary()) {
```

Comment:
Do we have tests for all the branches?

Reply:
We're working through adding the tests in this issue - https://github.com/opensearch-project/OpenSearch/issues/2355 - before we merge the feature branch back into main. This change was just to remove the failure/warning.